cosmos/launchpad/util.go

```diff
@@ -0,0 +1,97 @@
+package launchpad
+
+import (
+	"context"
+	"strconv"
+
+	cosmosclient "github.com/tendermint/cosmos-rosetta-gateway/cosmos/launchpad/client/sdk/generated"
+	tendermintclient "github.com/tendermint/cosmos-rosetta-gateway/cosmos/launchpad/client/tendermint/generated"
+
+	"github.com/coinbase/rosetta-sdk-go/types"
+)
+
+// getTxByHash calls
+func (l Launchpad) getTxByHash(ctx context.Context, hash string) (*types.Transaction, *types.Error) {
+	txQuery, _, err := l.cosmos.Transactions.TxsHashGet(ctx, hash)
+	if err != nil {
+		return nil, ErrNodeConnection
+	}
+
+	tx := cosmosTxToRosettaTx(txQuery)
+
+	return tx, nil
+}
+func toBlockIdentifier(result tendermintclient.BlockComplete) (*types.BlockIdentifier, error) {
+	if result.BlockId.Hash == "" {
+		return nil, nil
+	}
+	height, err := strconv.ParseUint(result.Block.Header.Height, 10, 64)
+	if err != nil {
+		return nil, err
+	}
+	return &types.BlockIdentifier{
+		Index: int64(height),
+		Hash:  result.BlockId.Hash,
+	}, nil
+}
+
+func toTransactions(txs []cosmosclient.TxQuery) (transactions []*types.Transaction, err error) {
+	for _, tx := range txs {
+		transactions = append(transactions, cosmosTxToRosettaTx(tx))
+	}
+	return
+}
+
+// tendermintTxToRosettaTx converts a Tendermint api TxResponseResult to a Transaction
+// in the type expected by Rosetta.
+func tendermintTxToRosettaTx(res tendermintclient.TxResponseResult) *types.Transaction {
+	return &types.Transaction{
+		TransactionIdentifier: &types.TransactionIdentifier{
+			Hash: res.Hash,
+		},
+		Operations: nil, // TODO difficult to get the operations from the mempool (maybe not worth it due to block times).
```

Comment:
Operations seems to be required for this API. Do you think we should keep this PR open until we figure out filling operations here?

https://github.com/coinbase/rosetta-specifications/blob/master/models/Transaction.yaml#L21

Reply:
I think we can leave the TODO comment, the problem I think is that here we just have the tx encoded. Due to our blocktimes I am not sure if it is worth it to solve this issue. Even rosetta does not force sometime the mempool calls.

https://www.rosetta-api.org/docs/all_methods.html#note-on-mempool-methods

So we can leave this like it is until really we have the time and maybe the necessity to implement it.